operations/mimir-mixin/dashboards/dashboard-utils.libsonnet

```diff
@@ -248,20 +248,31 @@ local utils = import 'mixin-utils/utils.libsonnet';
       },
     },
 
-  resourcesPanelLegend(first_legend)::
+  resourceUtilizationAndLimitLegend(resourceName)::
     if $._config.deployment_type == 'kubernetes'
-    then [first_legend, 'limit', 'request']
+    then [resourceName, 'limit', 'request']
     // limit and request does not makes sense when running on baremetal
-    else [first_legend],
+    else [resourceName],
 
-  resourcesPanelQueries(metric, instanceName)::
+  resourceUtilizationQuery(metric, instanceName)::
     if $._config.deployment_type == 'kubernetes'
-    then [
+    then
+      $._config.resources_panel_queries[$._config.deployment_type]['%s_usage' % metric] % {
+        instance: $._config.per_instance_label,
+        namespace: $.namespaceMatcher(),
+        instanceName: instanceName,
+      }
```

Comment:
Isn't this branch of the `if` the same as the other? Why not remove the `if` altogether?

Reply:
Oh right! I refactored it without noticing it was the exact same. Fixed in https://github.com/grafana/mimir/pull/3481/commits/f8c95c24aed8be7179d6662f2cd1433c49759970.